### DISCUSSION TITLE
How to create relay node simply?

### DISCUSSION THREAD

**Discussion by dohaicuong** [hayes/pothos#771] [Q&A] [answered]
Hi
I can see relay plugin allows me to create node https://pothos-graphql.dev/docs/plugins/relay#creating-nodes
and this plugin to create simpleObject with interfaces https://pothos-graphql.dev/docs/plugins/simple-objects

How can I create simpleObject with relay node interface from relay plugin?

**Comment by hayes** [comment, marked as answer]
This isn't really supported right now.  I would recommend just using the normal way of defining relay nodes.  Is there something you are having trouble defining with the that method?

**Reply by hayes** [reply]
Looking at the docs, I just noticed it only shows usage with a class, which is probably the issue.  You can just use a string as the first argument, builder.node should be able to infer types from the load methods

**Reply by dohaicuong** [reply]
Yeah I was able to follow the docs using class.
Just wondering if there are other ways to do that without class or best to use simpleObject plugin to do so :))

I just tried what you suggest by using the first `builder.node` argument as string instead of Class, it's working.
However typescript is not happy :\
on the `builder.node` it would throw
```
Argument of type 'string' is not assignable to parameter of type 'ObjectParam<ExtendDefaultTypes<SchemaBuilderContext>>'
```

and for the queryField that I return that type would throwing some typescript error as well since it unable to infer the types `PothosSchemaTypes.ObjectRef<any, any>`

**Reply by dohaicuong** [reply]
The best I can make it works and typescript happy is
define the type
```
type User = {
  id: string
  email: string
}
```

and tell the builder.node
```
export const UserType: PothosSchemaTypes.ObjectRef<User, User> = builder.node('User' as any, {})
````

it seems the problem is it's unable to infer the type from load method

**Reply by hayes** [reply]
This was my mistake, sorry.  I mixed up the APIs for the relay and dataloader plugins.

What you can do instead here is:
```ts
type User = {
  id: string
  email: string
}

export const UserType = builder.objectRef<User>('User')

builder.node(UserType, { ... })
```

**Reply by hayes** [reply]
this should be better documented better

**Reply by dohaicuong** [reply]
Cool thank you
